src/main/kotlin/com/example/taskManagerController/controller/TaskController.kt

```
@@ -0,0 +1,52 @@
+        } catch (e: IllegalArgumentException) {
+            ResponseEntity.badRequest().build()
+        }
+    }
+
+    @PutMapping("/{id}")
+    fun updateTask(@PathVariable id: UUID, @RequestBody updatedTask: Task): ResponseEntity<Task> {
+        return try {
+            taskService.updateTask(id, updatedTask)
+                ?.let { task -> ResponseEntity.ok(task) }
+                ?: ResponseEntity.notFound().build()
+        } catch (e: IllegalArgumentException) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)